### PR TITLE
ci: make postgres container ID non-deterministic

### PR DESCRIPTION
**Pull request by michaelbeaumont**
Might [this be the reason](https://pkg.go.dev/math/rand#pkg-overview) we still see [flakiness in the postgres suite](https://app.circleci.com/pipelines/github/kumahq/kuma/16207/workflows/124126f7-9a65-4df6-8a49-f85b1ed7fdbc/jobs/215132/tests) after https://github.com/kumahq/kuma/pull/4668?

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch)?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by bartsmykla — APPROVED** [review]
lgtm

File: pkg/test/store/postgres/test_container.go [hunk 1]
```diff
@@ -40,17 +41,22 @@ func resourceDir() string {
...
 	if v.WithTLS {
 		mode = "tls"
 	}
-	uniqueId := strconv.Itoa(rand.Int())
 	buildArgs["MODE"] = &mode
+
 	// Add a uniqueId to make each image different, this was causing flakiness as test-container
 	// deletes the image that it builds unconditionally during teardown and fails if the image doesn't exist.
 	// In the case of parallel tests it's possible that the same image was used in multiple tests and the second teardown would fail.
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
```
Comment (lahabana) [inline pkg/test/store/postgres/test_container.go:56]: Maybe use: `GinkgoRandomSeed()` (This seed is in the log, makes it reproduceable still).

**Review by lahabana — CHANGES_REQUESTED** [review]

**Comment by lahabana** [comment]
@michaelbeaumont Could you explain some more why you think this may solve the issue? You think the random generator returns twice the same number (which actually is possible :) )?

**Comment by michaelbeaumont** [comment]
@lahabana I suspect it returns the same random number for every parallel process:

> Top-level functions, such as Float64 and Int, use a default shared Source that produces a deterministic sequence of values each time a program is run

**Review by lahabana — APPROVED** [review]

File: pkg/test/store/postgres/test_container.go [hunk 2]
```diff
@@ -40,17 +41,22 @@ func resourceDir() string {
...
 	if v.WithTLS {
 		mode = "tls"
 	}
-	uniqueId := strconv.Itoa(rand.Int())
 	buildArgs["MODE"] = &mode
+
 	// Add a uniqueId to make each image different, this was causing flakiness as test-container
 	// deletes the image that it builds unconditionally during teardown and fails if the image doesn't exist.
 	// In the case of parallel tests it's possible that the same image was used in multiple tests and the second teardown would fail.
+	r := rand.New(rand.NewSource(int64(GinkgoParallelProcess())))
```
Comment (lahabana) [inline pkg/test/store/postgres/test_container.go:56]: Aren't test inside a ParallelProcess ran sequentially? In which case the uniqueId being the output of: GinkgoParallelProcess might be sufficient

Comment (michaelbeaumont) [reply pkg/test/store/postgres/test_container.go]: Yeah of course

**Pull request merged**